test/affinity_assistant_test.go

```diff
@@ -138,3 +132,127 @@ spec:
+	pr := parse.MustParseV1PipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  pipelineSpec:
+    workspaces:
+    - name: my-workspace
+    - name: my-workspace2
+    tasks:
+    - name: foo
+      workspaces:
+      - name: my-workspace
+      taskSpec:
+        steps:
+        - image: busybox
+          script: echo hello foo
+    - name: bar
+      workspaces:
+      - name: my-workspace2
+      taskSpec:
+        steps:
+        - image: busybox
+          script: echo hello bar
+    - name: double-ws
+      workspaces:
+      - name: my-workspace
+      - name: my-workspace2
+      taskSpec:
+        steps:
+        - image: busybox
+          script: echo double-ws
+    - name: no-ws
+      taskSpec:
+        steps:
+        - image: busybox
+        script: echo no-ws
+  workspaces:
+  - name: my-workspace
+    volumeClaimTemplate:
+      metadata:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 16Mi
+  - name: my-workspace2
+    volumeClaimTemplate:
+      metadata:
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 16Mi
+`, prName, namespace))
+
+	// create PipelineRun
+	if _, err := c.V1PipelineRunClient.Create(ctx, pr, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create PipelineRun: %s", err)
+	}
+
+	// wait for PipelineRun to finish
+	t.Logf("Waiting for PipelineRun in namespace %s to finish", namespace)
+	if err := WaitForPipelineRunState(ctx, c, prName, timeout, PipelineRunSucceed(prName), "PipelineRunSucceeded", v1Version); err != nil {
+		t.Errorf("Error waiting for PipelineRun to finish: %s", err)
+	}
+
+	// validate PipelineRun pods sharing the same PVC are scheduled to the same node
+	podNames := []string{fmt.Sprintf("%v-foo-pod", prName), fmt.Sprintf("%v-bar-pod", prName), fmt.Sprintf("%v-double-ws-pod", prName), fmt.Sprintf("%v-no-ws-pod", prName)}
+	validatePodAffinity(t, ctx, podNames, namespace, c.KubeClient)
+}
+
+// validatePodAffinity checks if all the given pods are scheduled to the same node
+func validatePodAffinity(t *testing.T, ctx context.Context, podNames []string, namespace string, client kubernetes.Interface) {
+	t.Helper()
+	nodeName := ""
+	for _, podName := range podNames {
+		pod, err := client.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Failed to get pod: %v, err: %v", podName, err)
+		}
+
+		if nodeName == "" {
+			nodeName = pod.Spec.NodeName
+		} else if pod.Spec.NodeName != nodeName {
+			t.Errorf("pods are not scheduled to the same node as expected %v vs %v", nodeName, pod.Spec.NodeName)
+		}
+	}
+}
+
+func resetFeatureFlagAndCleanup(ctx context.Context, t *testing.T, c *clients, namespace string) {
+	t.Helper()
+	configMapData := map[string]string{
+		"coschedule":                 config.CoscheduleWorkspaces,
+		"disable-affinity-assistant": "false",
```

Comment:
is there any way you could use the variables representing default values here rather than hard-coding them, so if we change the default values we don't also have to update this test?

Reply:
Thanks! Changed to default value variables